### PR TITLE
shouldn't try leader first when leader's store is slow in prefer-leader strategy in replicaSelectorV2

### LINKED ISSUE #1214: shouldn't try leader first when leader's store is slow in prefer-leader strategy in replicaSelectorV2
# Issue

### 1. Minimal reproduce step

Make config `EnableForwarding` set to true, which is enable forwarding.

In the following case, suppose the region leader is in `store1`, and `store1` is been marked slow since got `ServerIsBusyErr` before.

```go
	ca = replicaSelectorAccessPathCase{
			reqType:   tikvrpc.CmdGet,
			readType:  kv.ReplicaReadPreferLeader,
			staleRead: false,
			timeout:   0,
			accessErr: []RegionErrorType{ServerIsBusyErr, ServerIsBusyErr, ServerIsBusyErr},
```

### 2. What did you expect to see?

```go
			expect: &accessPathResult{
				accessPath: []string{
					"{addr: store2, replica-read: true, stale-read: false}", // won't try leader in store1, since it is slow.
					"{addr: store3, replica-read: true, stale-read: false}",
					"{addr: store1, replica-read: true, stale-read: false}",
				},
				respErr:         "",
				respRegionError: fakeEpochNotMatch,
				backoffCnt:      1,
				backoffDetail:   []string{"tikvServerBusy+1"},
				regionIsValid:   false,
			},
```

### 3. What did you see instead?

```go
			expect: &accessPathResult{
				accessPath: []string{
					"{addr: store1, replica-read: true, stale-read: false}", 
					"{addr: store2, replica-read: true, stale-read: false}",
					"{addr: store3, replica-read: true, stale-read: false}",
				},
				respErr:         "",
				respRegionError: fakeEpochNotMatch,
				backoffCnt:      1,
				backoffDetail:   []string{"tikvServerBusy+1"},
				regionIsValid:   false,
			},
```

### PR DESCRIPTION
**Pull request by crazycs520**
close #1214

### CODE REVIEW

**Review by MyonKeminta — APPROVED** [review]

File: internal/locate/replica_selector_test.go [hunk 1]
```diff
@@ -1658,6 +1646,49 @@ func TestReplicaReadAccessPathByMixedAndPreferLeaderCase(t *testing.T) {
...
+			readType:  kv.ReplicaReadPreferLeader,
+			staleRead: false,
+			timeout:   0,
+			accessErr: []RegionErrorType{ServerIsBusyErr, ServerIsBusyErr, ServerIsBusyErr},
+			beforeRun: func() { /* don't resetStoreState */ },
+			expect: &accessPathResult{
+				accessPath: []string{
+					"{addr: store2, replica-read: true, stale-read: false}", // won't try leader in store1, since it is slow.
+					"{addr: store3, replica-read: true, stale-read: false}",
+					"{addr: store1, replica-read: true, stale-read: false}",
```
Comment (you06) [inline internal/locate/replica_selector_test.go:1681]: I'm wondering what will be if store2 is also slow. I think the access path shall be `store3 -> store1 -> store2`, but v1 seems don't try store2 after store1.
Is it an advantage of v2?

**Review by you06 — APPROVED** [review]
rest LGTM

File: internal/locate/replica_selector_test.go (hunk 1 above)
Comment (cfzjywxk) [reply internal/locate/replica_selector_test.go:1681]: @crazycs520 PTAL

Comment (crazycs520) [reply internal/locate/replica_selector_test.go:1681]: Nice catch, It looks like v2 is not compatible with v1 in this case, I'm going to modify v2 to be compatible with v1 and add tests for that.

Comment (crazycs520) [reply internal/locate/replica_selector_test.go:1681]: @you06 addressed, and add a test for it, PTAL.

**Review by you06 — APPROVED** [review]

**Review by cfzjywxk — APPROVED** [review]

**Pull request merged**